Lib/ntpath.py

```diff
@@ -727,17 +722,17 @@ def realpath(path, *, strict=False):
             new_unc_prefix = b'\\\\'
             cwd = os.getcwdb()
             # bpo-38081: Special case for realpath(b'nul')
-            if normcase(path) == normcase(os.fsencode(devnull)):
+            if normcase(path) == b'nul':
                 return b'\\\\.\\NUL'
         else:
             prefix = '\\\\?\\'
             unc_prefix = '\\\\?\\UNC\\'
             new_unc_prefix = '\\\\'
             cwd = os.getcwd()
             # bpo-38081: Special case for realpath('nul')
-            if normcase(path) == normcase(devnull):
+            if normcase(path) == 'nul':
```

Comment:
> Strings representing various path-related bits and pieces.
> These are primarily for export; internally, they are **hardcoded**.

But if you would prefer it to be assigned first, that would be a trivial change.

Reply:
I don't personally take that comment to mean "We should never refer to these symbols internally within this module; we should always use magic strings instead"

Reply:
I think the code will be uglier with `_devnull_bytes`. It will also be inconsistent because only this function will use the global constants. And it will break the function if these constants are monkeypatched.
I think assigning them first is a good middle ground between readability and consistency.